app/screens/edit_channel/edit_channel.js

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {defineMessages, injectIntl, intlShape} from 'react-intl';
+import {
+    Keyboard,
+    InteractionManager
+} from 'react-native';
+
+import ChannelInfo from 'app/components/channel_info';
+import {RequestStatus} from 'mattermost-redux/constants';
+import EventEmitter from 'mattermost-redux/utils/event_emitter';
+import {ViewTypes} from 'app/constants';
+
+import {cleanUpUrlable} from 'app/utils/url';
+
+const holders = defineMessages({
```

Comment:
I'm not really a fan of this, defineMessages adds unnecessary boilerplate, you could either define your messages as 
```js
cons messages = {
    display_name_required: {
        id: '...',
       defaultMessage: '...'
    },
    ....
};
```

or simply do `intl.formatMessage({id: '...', defaultMessage: '...'}, values})` and values is optional